src/cmd_processor.cc

```diff
@@ -0,0 +1,46 @@
+#include "cmd_processor.h"
+
+#include <libqalculate/Calculator.h>
+
+namespace bb
+{
+    class CMDProcessor::Impl
+    {
+    public:
+        Impl();
+        std::string process(const std::string &cmd);
+
+        Calculator calc;
+    };
+}
+
+// Internal implementation
+
+bb::CMDProcessor::Impl::Impl()
+    : calc(Calculator())
+{
+    calc.useIntervalArithmetic(false);
+    calc.loadExchangeRates();
+    calc.loadGlobalCurrencies();
+    calc.loadGlobalDefinitions();
+    calc.loadLocalDefinitions();
+}
+
+std::string bb::CMDProcessor::Impl::process(const std::string &cmd)
+{
+    return calc.calculateAndPrint(calc.unlocalizeExpression(cmd));
```

Comment:
The previous comment was marked as outdated but I still have some concerns with this line. For example, is this line running in a different thread? how can be aborted? how does the timeout work?

Reply:
https://github.com/Qalculate/libqalculate/issues/148